--- linux-4.9.20/drivers/net/phy/ar40xx.c.orig	2017-05-02 17:59:11.827218985 +0200
+++ linux-4.9.20/drivers/net/phy/ar40xx.c	2017-05-02 18:06:44.358165050 +0200
@@ -2008,6 +2008,42 @@ static int ar40xx_probe(struct platform_
 		return -EIO;
 	}

+	if (of_property_read_bool(switch_node, "single_nic")) {
+#define ESS_RGMII_CTRL		0x0004
+
+/* Port status registers */
+#define ESS_PORT0_HEADER_CTRL	0x009C
+#define ESS_PORT0_STATUS	0x007C
+#define ESS_PORT1_STATUS	0x0080
+#define ESS_PORT2_STATUS	0x0084
+#define ESS_PORT3_STATUS	0x0088
+#define ESS_PORT4_STATUS	0x008C
+#define ESS_PORT5_STATUS	0x0090
+
+#define ESS_PORT_STATUS_HDX_FLOW_CTL	0x80
+#define ESS_PORT_STATUS_DUPLEX_MODE	0x40
+#define ESS_PORT_STATUS_RX_FLOW_EN	0x20
+#define ESS_PORT_STATUS_TX_FLOW_EN	0x10
+#define ESS_PORT_STATUS_RX_MAC_EN	0x08
+#define ESS_PORT_STATUS_TX_MAC_EN	0x04
+#define ESS_PORT_STATUS_SPEED_INV	0x03
+#define ESS_PORT_STATUS_SPEED_1000	0x02
+#define ESS_PORT_STATUS_SPEED_100	0x01
+#define ESS_PORT_STATUS_SPEED_10	0x00
+
+#define ESS_PORT_1G_FDX      (ESS_PORT_STATUS_DUPLEX_MODE | ESS_PORT_STATUS_RX_FLOW_EN | \
+                              ESS_PORT_STATUS_TX_FLOW_EN  | ESS_PORT_STATUS_RX_MAC_EN  | \
+                              ESS_PORT_STATUS_TX_MAC_EN   | ESS_PORT_STATUS_SPEED_1000)
+
+		if (priv->mac_mode == 0x3) {
+			ar40xx_write(priv, ESS_RGMII_CTRL, 0x400);
+		}
+		ar40xx_write(priv, ESS_PORT0_STATUS, ESS_PORT_1G_FDX);
+		ar40xx_write(priv, ESS_PORT5_STATUS, ESS_PORT_1G_FDX);
+		ar40xx_write(priv, ESS_PORT0_HEADER_CTRL, 0);
+		return -ENODEV;
+	}
+
 	ret = phy_driver_register(&ar40xx_phy_driver, THIS_MODULE);
 	if (ret) {
 		dev_err(&pdev->dev, "Failed to register ar40xx phy driver!\n");
